substrate/primitives/keyring/Cargo.toml

```diff
@@ -14,12 +14,27 @@ readme = "README.md"
 targets = ["x86_64-unknown-linux-gnu"]
 
 [dependencies]
-lazy_static = "1.4.0"
+lazy_static = { version = "1.4.0", features = [ "spin_no_std" ], default-features = false }
 strum = { version = "0.24.1", features = ["derive"], default-features = false }
-sp-core = { path = "../core" }
-sp-runtime = { path = "../runtime" }
+sp-core = { path = "../core", default-features = false }
+sp-runtime = { path = "../runtime", default-features = false }
+sp-std = { path = "../std", default-features = false  }
+
+#todo: remove this (required for `cargo build --no-default-features`)
+sp-io = { path = "../io", default-features = false, optional = true }
+sp-runtime-interface = { path = "../runtime-interface", default-features = false, optional = true }
 
 [features]
+default = [ "std" ]
+std = [
+	"sp-core/std",
+	"sp-runtime/std",
+	"sp-std/std",
+]
+
+#todo: remove this (required for `cargo build --no-default-features`)
```

Comment:
`cargo check --target wasm32-unknown-unknown --no-default-features` would not require this.